### ISSUE TITLE
[TTNN]: ttnn.gelu_bw(approximate="none") composite decomposition has severe ULP errors (Max ULP = 32,460)

### ISSUE THREAD

**Issue by epam-iaroslav-voitovych** [tenstorrent/tt-metal#38643]
### Component / Area

TTNN / Eltwise Operations / Unary Backward / GELU Backward

### Issue Type

Bad Outputs

### Observed

`ttnn.gelu_bw(approximate="none")` — the main (non-experimental) API — has severe accuracy issues affecting approximately 50% of the BF16 input range:

1. **Wrong sign outputs** at x ≈ -3.7: derivative should be negative but returns positive
2. **High ULP errors** in deep negative (x < -5) and large positive (x > 5) regions
3. **10.69% of all BF16 values have ULP > 1000**
4. **Max ULP = 32,460** (expected: ≤ 1)

This is the **same mathematical bug** as issue #35971, but in a different code path.

### Implementation Details

The main API `ttnn.gelu_bw(approximate="none")` is implemented as a **composite host-side decomposition** in `ttnn/cpp/ttnn/operations/eltwise/unary_backward/unary_backward.cpp` (lines ~1613-1632). It dispatches approximately 10 separate TTNN sub-operations:

```cpp
float kAlpha = M_SQRT1_2;
float kBeta = M_2_SQRTPI * M_SQRT1_2 * 0.5;
Tensor cdf = ttnn::multiply(
    (ttnn::add(
        ttnn::erf(ttnn::multiply(input, kAlpha, ...)),
        1, ...)),
    0.5);
Tensor pdf = ttnn::multiply(
    ttnn::exp(ttnn::multiply(ttnn::multiply(input, input), -0.5), ...),
    kBeta, ...);
ttnn::multiply(grad, ttnn::add(cdf, ttnn::multiply(input, pdf)), ...);
```

**Root cause:** The formula `cdf = 0.5 * (1 + erf(x/√2))` suffers from catastrophic cancellation when `erf(x/√2)` approaches -1 for large negative x. The SFPU `erf()` implementation saturates, and `1 + (-1)` loses all significant digits. This cannot be fixed by rearranging the composite ops — the fundamental limitation is in the erf/erfc SFPU functions.

### Distinction from Issue #35971

Issue #35971 covers `ttnn.experimental.gelu_bw()` — a **fused device kernel** implementation (`eltwise_bw_gelu_approx_none.cpp`) that computes the entire GELU derivative in a single kernel. That was fixed by PR #36366 using piecewise polynomial approximation (Max ULP = 1).

This issue covers `ttnn.gelu_bw()` — the **main public API** that uses composite host-side decomposition. These are independent code paths. The main API does NOT call the experimental API.

| API | Implementation | Code Path | Status |
|-----|---------------|-----------|--------|
| `ttnn.experimental.gelu_bw()` | Fused device kernel | `ttnn/cpp/ttnn/operations/experimental/unary_backward/gelu_backward/device/kernels/compute/eltwise_bw_gelu_approx_none.cpp` | **Fixed** (PR #36366) |
| `ttnn.gelu_bw()` | Composite decomposition | `ttnn/cpp/ttnn/operations/eltwise/unary_backward/unary_backward.cpp` | **This issue** |

### Note on tanh mode

Both APIs also support `approximate="tanh"`, which uses the tanh-based GELU approximation matching PyTorch's `torch.nn.functional.gelu(approximate='tanh')`. The tanh mode is a distinct mathematical formulation and is **not affected** by this bug.

### Expected

GELU backward should compute `GELU'(x) = Φ(x) + x·φ(x)` where Φ is the standard normal CDF and φ is the standard normal PDF. Results should match the fp64 reference within 1 ULP for BF16.

### Reproducer

```python
import ttnn
import torch
import math

device = ttnn.open_device(0)

x_val = -3.7  # Wrong sign occurs here
input_tensor = ttnn.from_torch(
    torch.full((1, 1, 32, 32), x_val, dtype=torch.bfloat16),
    layout=ttnn.TILE_LAYOUT, device=device
)
grad_tensor = ttnn.from_torch(
    torch.full((1, 1, 32, 32), 1.0, dtype=torch.bfloat16),
    layout=ttnn.TILE_LAYOUT, device=device
)

# Main API (NOT experimental)
result = ttnn.gelu_bw(grad_tensor, input_tensor, approximate="none")
actual = ttnn.to_torch(result[0])[0, 0, 0, 0].item()

# Reference
SQRT2 = math.sqrt(2.0)
INV_SQRT_2PI = 1.0 / math.sqrt(2.0 * math.pi)
cdf = 0.5 * math.erfc(-x_val / SQRT2)
pdf = math.exp(-0.5 * x_val * x_val) * INV_SQRT_2PI
expected = cdf + x_val * pdf

print(f"x = {x_val}")
print(f"Expected: {expected:.6e} (NEGATIVE)")
print(f"Actual:   {actual:.6e} (POSITIVE - WRONG SIGN!)")

ttnn.close_device(device)
```

### Proposed Fix

Create a new fused device kernel under `ttnn/cpp/ttnn/operations/eltwise/unary_backward/` that calls `gelu_derivative_tile()` — the same polynomial SFPU function used by the experimental fix (PR #36366). This replaces the composite decomposition with a single device kernel while keeping the main API independent of the experimental namespace.

The shared SFPU infrastructure (`ckernel_sfpu_gelu.h`, `gelu.h`) provides the polynomial math. Only a new compute kernel and program factory need to be created under the main op.

### Frequency

100% reproducible. Affects all BF16 values in the broken regions.

### Priority

P2 (P1 consumer tt-train already uses the fixed experimental path)

### Impact

- Any user calling `ttnn.gelu_bw()` directly (not through tt-train) gets wrong gradients
- Wrong sign causes gradients to flow in the wrong direction during backpropagation
- The moderate negative region [-5, -2] contains commonly encountered activation values

### Related Issues

- #35971: Same bug in `ttnn.experimental.gelu_bw()` (fixed by PR #36366)
- #35290: ttnn.gelu() ULP errors (forward GELU, similar erf precision issue)
